pkg/controller/networkpolicy/networkpolicy.go

```diff
@@ -305,3 +296,111 @@ func NetworkPolicyForService(req router.Request, resp router.Response) error {
 
 	return nil
 }
+
+// PolicyForBuilder creates a Kubernetes NetworkPolicy to allow traffic to the buildkitd pods from the acorn-api only.
+// It also only allows outgoing traffic to CoreDNS, the Acorn registry, and the Internet.
+func PolicyForBuilder(req router.Request, resp router.Response) error {
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	} else if !*cfg.NetworkPolicies {
+		return nil
+	}
+
+	deployment := req.Object.(*appsv1.Deployment)
+	if deployment.Name != "buildkitd" && !strings.HasPrefix(deployment.Name, "bld-") {
+		// this is not a builder deployment
+		return nil
+	}
+
+	podLabels := deployment.Spec.Template.ObjectMeta.Labels
+	externalIPBlock, err := buildExternalIPBlock(req)
+	if err != nil {
+		return err
+	}
+
+	// build the NetPol
+	resp.Objects(&networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment.Name,
+			Namespace: deployment.Namespace,
+			Labels: map[string]string{
+				labels.AcornManaged: "true",
+			},
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: podLabels,
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
```

Comment:
Another usecase for builder is that we can publish builder using ingress and the build will just using ingress traffic. Hub workload cluster uses this mode. Have we accounted for allowing ingress traffic in this case?

Reply:
I was looking at our dev sandbox workload cluster and did not see any ingress resources exposing the builders. Do you know of a cluster where we are doing this so that I can look at it for reference?

Reply:
Sandbox workload cluster should have it. https://github.com/acorn-io/hub-system-charts/blob/main/templates/acorn-install.yaml#L53-L55

You can also try installing acorn using this flag --publishBuilders

Reply:
I tested this out. Thankfully, the `ForIngress` handler steps in and takes care of what we need to ensure that the ingress controller can proxy traffic to the builder without any issues.